### ISSUE TITLE
Crashes on startup relating to alsa and number of pcm devices

### ISSUE THREAD

**Issue by dsjoho** [xournalpp/xournalpp#3531]
**Affects versions :**
 - OS: Linux, arch linux
 - (Linux only) Desktop environment: Gnome
 - Which version of libgtk do you use
     - 3.24.30
 - Version of Xournal++: [e.g. 1.0.16, specific commit, specific branch]
     - 1.1.0
 - Installation method: official repositories

**Describe the bug**
Xournalpp crashes upon start if multiple alsa input/outputs are defined

**To Reproduce**
1. Have an incorrect alsa configuration, or more than 6 input/output channels in .asoundrc
2. Start xournalpp
3. Xournalpp crashes

**Expected behavior**
I expect xournalpp to be able to start no matter what my alsa configuration is. Sound is not mandatory for it?

**Additional context**
I don't know if my alsa configuration is faulty or not, but I can say this: I can play audio through pulseaudio normally, and I can start qjackctl and play audio through my desired routings. 
So no other programs have issues with my setup.

My .asoundrc contains entries like below,
```
pcm.outch8 {
    type dshare
    ipc_key 1111
    slave cardref
    bindings [ 7 ]
}

pcm.inch8 {
    type dsnoop
    ipc_key 1111
    slave cardref
    bindings [ 7 ]
}
```
From where the number goes 1-8, but similar contents of each pcm entry.
**If I comment nr 6-8, xournalpp runs. If I uncomment 6,7 or 8 xournalpp crashes on startup.**

Terminal log:
```
xournalpp
** Message: 11:40:19.231: TEXTDOMAINDIR = (null), PACKAGE_LOCALE_DIR = /usr/share/locale, chosen directory = /usr/share/locale

** (xournalpp:50977): WARNING **: 11:40:19.232: Unsupported input method: xim, changed to: ibus
ALSA lib pcm_dsnoop.c:600:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1035:(snd_pcm_dmix_open) unable to open slave
ALSA lib confmisc.c:1369:(snd_func_refer) Unable to find definition 'cards.0.pcm.rear.0:CARD=0'
ALSA lib conf.c:5111:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5599:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM rear
ALSA lib confmisc.c:1369:(snd_func_refer) Unable to find definition 'cards.0.pcm.center_lfe.0:CARD=0'
ALSA lib conf.c:5111:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5599:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM center_lfe
ALSA lib confmisc.c:1369:(snd_func_refer) Unable to find definition 'cards.0.pcm.side.0:CARD=0'
ALSA lib conf.c:5111:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5599:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM side
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib pcm_oss.c:397:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib pcm_oss.c:397:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib pcm_usb_stream.c:482:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib pcm_usb_stream.c:482:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib pcm_dshare.c:701:(snd_pcm_dshare_open) The dshare plugin supports only playback stream
ALSA lib pcm_dsnoop.c:600:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dsnoop.c:574:(snd_pcm_dsnoop_open) The dsnoop plugin supports only capture stream
Date: Sun Oct 31 11:40:19 2021
Error: signal 11
[bt]: (0) xournalpp(+0x1e044d) [0x56511976244d]
[bt]: (1) /usr/lib/libc.so.6(+0x3cda0) [0x7f6f49cb5da0]
[bt]: (2) /usr/lib/libasound.so.2(snd_pcm_area_silence+0x1a0) [0x7f6f48c1b210]
[bt]: (3) /usr/lib/libasound.so.2(+0x86d81) [0x7f6f48c4fd81]
[bt]: (4) /usr/lib/libasound.so.2(+0x873d8) [0x7f6f48c503d8]
[bt]: (5) /usr/lib/libasound.so.2(snd_pcm_close+0x76) [0x7f6f48c1a846]
[bt]: (6) /usr/lib/libportaudio.so.2(+0xe1cf) [0x7f6f4a2911cf]
[bt]: (7) /usr/lib/libportaudio.so.2(+0xe89e) [0x7f6f4a29189e]
[bt]: (8) /usr/lib/libportaudio.so.2(+0x1396c) [0x7f6f4a29696c]
[bt]: (9) /usr/lib/libportaudio.so.2(PaAlsa_Initialize+0x5fa) [0x7f6f4a29725a]
[bt]: (10) /usr/lib/libportaudio.so.2(Pa_Initialize+0x106) [0x7f6f4a28a336]
[bt]: (11) /usr/lib/libportaudiocpp.so.0(_ZN9portaudio6System10initializeEv+0x53) [0x7f6f4a2be2e3]
[bt]: (12) xournalpp(_ZN15AudioControllerC1EP8SettingsP7Control+0x2c) [0x56511963d43c]
[bt]: (13) xournalpp(_ZN7ControlC1EP13_GApplicationP15GladeSearchpath+0x4a2) [0x565119635462]
[bt]: (14) xournalpp(+0xcfd94) [0x565119651d94]
[bt]: (15) /usr/lib/libgobject-2.0.so.0(g_closure_invoke+0x16f) [0x7f6f4a4e9d8f]
[bt]: (16) /usr/lib/libgobject-2.0.so.0(+0x2e588) [0x7f6f4a505588]
[bt]: (17) /usr/lib/libgobject-2.0.so.0(g_signal_emit_valist+0xe79) [0x7f6f4a506...skipping...
Date: Sun Oct 31 11:40:19 2021
Error: signal 11
[bt]: (0) xournalpp(+0x1e044d) [0x56511976244d]
[bt]: (1) /usr/lib/libc.so.6(+0x3cda0) [0x7f6f49cb5da0]
[bt]: (2) /usr/lib/libasound.so.2(snd_pcm_area_silence+0x1a0) [0x7f6f48c1b210]
[bt]: (3) /usr/lib/libasound.so.2(+0x86d81) [0x7f6f48c4fd81]
[bt]: (4) /usr/lib/libasound.so.2(+0x873d8) [0x7f6f48c503d8]
[bt]: (5) /usr/lib/libasound.so.2(snd_pcm_close+0x76) [0x7f6f48c1a846]
[bt]: (6) /usr/lib/libportaudio.so.2(+0xe1cf) [0x7f6f4a2911cf]
[bt]: (7) /usr/lib/libportaudio.so.2(+0xe89e) [0x7f6f4a29189e]
[bt]: (8) /usr/lib/libportaudio.so.2(+0x1396c) [0x7f6f4a29696c]
[bt]: (9) /usr/lib/libportaudio.so.2(PaAlsa_Initialize+0x5fa) [0x7f6f4a29725a]
[bt]: (10) /usr/lib/libportaudio.so.2(Pa_Initialize+0x106) [0x7f6f4a28a336]
[bt]: (11) /usr/lib/libportaudiocpp.so.0(_ZN9portaudio6System10initializeEv+0x53
) [0x7f6f4a2be2e3]
[bt]: (12) xournalpp(_ZN15AudioControllerC1EP8SettingsP7Control+0x2c) [0x5651196
3d43c]
[bt]: (13) xournalpp(_ZN7ControlC1EP13_GApplicationP15GladeSearchpath+0x4a2) [0x
565119635462]
[bt]: (14) xournalpp(+0xcfd94) [0x565119651d94]
[bt]: (15) /usr/lib/libgobject-2.0.so.0(g_closure_invoke+0x16f) [0x7f6f4a4e9d8f]
[bt]: (16) /usr/lib/libgobject-2.0.so.0(+0x2e588) [0x7f6f4a505588]
[bt]: (17) /usr/lib/libgobject-2.0.so.0(g_signal_emit_valist+0xe79) [0x7f6f4a506
c49]
[bt]: (18) /usr/lib/libgobject-2.0.so.0(g_signal_emit+0x90) [0x7f6f4a5071a0]
[bt]: (19) /usr/lib/libgio-2.0.so.0(g_application_register+0x246) [0x7f6f4a5fcae6]
[bt]: (20) /usr/lib/libgio-2.0.so.0(+0xcc4ec) [0x7f6f4a5fd4ec]
[bt]: (21) /usr/lib/libgio-2.0.so.0(g_application_run+0x13a) [0x7f6f4a5fd84a]
[bt]: (22) xournalpp(_ZN11XournalMain3runEiPPc+0x5a3) [0x56511964eab3]
[bt]: (23) /usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7f6f49ca0b25]
[bt]: (24) xournalpp(_start+0x2e) [0x565119628f7e]


Try to get a better stracktrace...
[bt] #1 xournalpp(+0x1e09f4) [0x5651197629f4]
??:0
[bt] #2 /usr/lib/libc.so.6(+0x3cda0) [0x7f6f49cb5da0]
??:0
[bt] #3 /usr/lib/libasound.so.2(snd_pcm_area_silence+0x1a0) [0x7f6f48c1b210]
??:0
[bt] #4 /usr/lib/libasound.so.2(+0x86d81) [0x7f6f48c4fd81]
??:0
[bt] #5 /usr/lib/libasound.so.2(+0x873d8) [0x7f6f48c503d8]
??:0
[bt] #6 /usr/lib/libasound.so.2(snd_pcm_close+0x76) [0x7f6f48c1a846]
ALSA lib pcm_dshare.c:701:(snd_pcm_dshare_open) The dshare plugin supports only playback stream
ALSA lib pcm_dsnoop.c:600:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dsnoop.c:574:(snd_pcm_dsnoop_open) The dsnoop plugin supports only capture stream
ALSA lib pcm_dshare.c:701:(snd_pcm_dshare_open) The dshare plugin supports only playback stream
ALSA lib pcm_dsnoop.c:600:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dsnoop.c:574:(snd_pcm_dsnoop_open) The dsnoop plugin supports only capture stream
ALSA lib pcm_dshare.c:701:(snd_pcm_dshare_open) The dshare plugin supports only playback stream

** (xournalpp:50977): WARNING **: 11:40:19.290: [Crash Handler] Crashed with signal 11

** (xournalpp:50977): WARNING **: 11:40:19.290: [Crash Handler] Wrote crash log to: /home/user/.cache/xournalpp/errorlogs/errorlog.20211031-114019.log
```
[errorlog.20211031-114019.log](https://github.com/xournalpp/xournalpp/files/7448585/errorlog.20211031-114019.log)

**Comment by the-drunk-coder** [comment]
Hi, I'm experiencing a similar problem. I tried the regular and git versions on Arch Linux. I'm using pipewire as a drop-in replacement for pulseaudio. Here's the error (terminal output):

```
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.front
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround21
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround21
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround40
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround41
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround50
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround51
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround71
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.hdmi
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.hdmi
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.modem
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.modem
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.phoneline
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.phoneline
xournalpp: src/hostapi/alsa/pa_linux_alsa.c:900: GropeDevice: Assertion `maxChans <= INT_MAX' failed.

** (xournalpp:46966): WARNING **: 13:11:49.528: [Crash Handler] Crashed with signal 6
```

**Comment by rolandlo** [comment]
> 
> Is there a way to start xournalpp without any audio ?

No, unfortunately not. You would have to touch the code to get it running without audio.

**Comment by the-drunk-coder** [comment]
So given that the original problem was related to plain pulseaudio, it might not be pipewire at all ...

**Comment by rolandlo** [comment]
Closed in view of #5068.
